lib/tools/system-calls.js

```diff
@@ -344,6 +345,23 @@ systemCallMethods.killAllEmulators = async function () {
   }
 };
 
+systemCallMethods.killEmulator = async function (avdName) {
+  let cmd;
+  if (system.isWindows()) {
+    cmd = `for /f "tokens=2 delims=," %a in ('tasklist /FI "imagename eq emulator*" /V /FO:csv /NH ^| findstr /r /c:".*:${avdName}.*"') do taskkill /pid %a /F`;
+  } else {
+    cmd = `pkill -9 -f 'avd ${avdName}|@${avdName}'`;
+  }
+  try {
+    log.debug(`killing avd '${avdName}'`);
+    await (B.promisify(cp.exec))(cmd);
+    await sleep(1000);
```

Comment:
sometimes emulator stock and this command will force kill it and there will be no log to depend on to check if it. If it is stocked at the start of emulator, there will be no device added in running devices to check kill result in there.

Since this method is for killing emulator under any circumstances, no log or method can be reliable for checking result of command.